src/amltk/pipeline/node.py

```diff
@@ -468,15 +468,34 @@ def linearized_fidelity(self, value: float) -> dict[str, int | float | Any]:
 
         return prefix_keys(d, f"{self.name}:")
 
-    def iter(self) -> Iterator[Node]:
-        """Iterate the the nodes, including this node.
+    def iter(self, skip_unchosen: bool = False) -> Iterator[Node]:
+        """Recursively iterate through the nodes starting from this node.
+
+        This method traverses the nodes in a depth-first manner, including
+        the current node and its children nodes.
+
+        Args:
+            skip_unchosen (bool): Flag to skip unchosen nodes in Choice nodes.
 
         Yields:
-            The nodes connected to this node
+            Iterator[Node]: Nodes connected to this node.
         """
+        # Import Choice node to avoid circular imports
+        from amltk.pipeline.components import Choice
+
+        # Yield the current node
         yield self
+
+        # Iterate through the child nodes
         for node in self.nodes:
-            yield from node.iter()
+            if skip_unchosen and isinstance(node, Choice):
+                # If the node is a Choice and skipping unchosen nodes is enabled
+                chosen_node = node.chosen()
+                if chosen_node:
+                    yield from chosen_node.iter(skip_unchosen)
```

Comment:
And if there was no `Node` chosen, i.e. `chosen_node == None`? Im not entirely sure what to do here but I imagine it should resort to iterating through all it's children? Open to suggestions why not

Reply:
Do you mean, what will be the consequence if a user didn't use the `Choice` class in the pipeline, but passed the `skip_unchosen=False`?